### ISSUE TITLE
End Turn

### ISSUE THREAD

**Issue by Shivanitanwar** [microsoft/BotFramework-Composer#8060]
## End Turn - Switch Statement

I have a scenario in my bot where I am calling a multi choice prompt and based on user Response I am using a Switch statement.
Inside the switch statement I am ending the turn.
So, I am expecting that it will move ahead in the dialog after ending the turn but the bot clearly stops after the turn is ended, doesn't execute anything beyond the end Turn in the dialog (it doesn't even go to parent dialog).

Can you please help me understand the End Turn Activity. 

## Version
1.4.1

## OS

- [ ] macOS
-  Windows
- [ ] Ubuntu

## To Reproduce

1. Create two dialogs - A(parent) and B(child)
2. Inside dialog B - create a prompt multi choice and then a switch statement based on the expected answers from multi choice.
3. Inside one of the branch of switch statement End Turn.
4. Send a response after ending turn.

## Expected behavior

Either the flow should move ahead in Dialog B and execute whatever is there in B after End Turn activity OR go to the Parent dialog. Theoretically, when using End Turn, I think the flow should move ahead in Dialog B itself. It goes to the parent only when the existing Dialog (child dialog) is ended by using End Dialog activity.

## Additional context

Trying to figure out what exactly to expect from End Turn and right place to use it?

**Comment by stevkan** [comment]
@Shivanitanwar, could you zip your bot (removing any keys, etc.) and post it here? When I attempt a repro following your steps, the flow runs as you have described in the expected behavior. The turn ends, user input advances the flow thru the rest of the switch statement (effectively exiting it), user input advances the flow beyond thru the rest of Dialog B, then it returns back to Dialog A (the parent dialog).

**Comment by Shivanitanwar** [comment]
@stevkan Thanks for the response.
where did you exactly end the turn (inside switch statement or outside) ?
I have attached this sample project for your reference.
[MultiChoice-Test.zip](https://github.com/microsoft/BotFramework-Composer/files/6662419/MultiChoice-Test.zip)

**Comment by stevkan** [comment]
@Shivanitanwar, thank you for your patience while I looked closer at this. So, when I run your sample, this is the flow I'm experiencing:

1. Greeting trigger is activated
1. Dialog_A sends initial response, "Calling Dialog B here"
1. Dialog_A calls Dialog_B
1. Dialog_B present user with choice option (**Yes** or **No**)
1. When **Yes** is selected
    1. Response is sent, "Testing in Yes branch"
    1. End turn is called which "pauses" the flow
    1. User input is entered, "unpausing" the flow
1. Dialog_B sends the dialog's last response, "After turn"
1. Dialog_B returns control back to Dialog_A
1. Dialog_A sends the dialog's last response, "After Dialog B - Inside A"
1. (Control is passed from Dialog_A to the parent dialog, however, as there are no other triggers setup, no other dialogs are called. No responses are given to any additional user input.)

You're initial post says "the bot clearly stops after the turn is ended." Are you saying that steps 6-8 never occur once the turn is ended? The End turn action is a way to pause the flow without exiting the dialog. The dialog remains on top of the memory stack waiting for additional input or a process to complete. Once user input is sent, the dialog _should_ pick up where it left off and, in your case, display the messages in steps 6-8.

It's also important to know that when a dialog is called with "[Begin a new dialog](https://docs.microsoft.com/en-us/composer/concept-dialog?tabs=v2x#dialog-actions)", once that dialog ends, control is passed back to the caller. In your sample, the parent dialog calls Dialog_A which then calls Dialog_B. As I mention in step 9, when Dialog_B finishes, control is passed back to Dialog_A which, once finished, then returns control back to the parent. But, as there are no other triggers setup from which to call another dialog, the bot does nothing. Any user input at this point  goes unanswered.

When you run your sample, after it calls End turn, are you able to enter any user inputs? If you are and the bot isn't responding, then I suspect you reached step 9 as described above.

In short, your sample is running as expected for me. End turn is called, the bot pauses, I enter a user response, and the bot continues where it left off sending both of the final responses from Dialog_B and then Dialog_A before returning control to the parent dialog.

**Comment by Shivanitanwar** [comment]
@Stevkan Thanks for your response. I was not sending any User Input once the turn was ended. I thought the flow would go ahead automatically and bot would send the response (without any user Input).

Basically it expects the User Input to resume the flow. Thanks @stevkan , this clarifies the behavior of End Turn for me now.

Just the last question I have in my mind and I am not sure if it is even possible -- Is there any other way to resume the flow after end turn (apart from User Input) ? 
Because the place where I am trying to use "End Turn" in my actual Bot, doesn't expect any User Input after the "End Turn" activity.

**Comment by stevkan** [comment]
@Shivanitanwar, there isn't any way that I'm aware of or could easily find. For clarity, what are you trying to accomplish by ending the turn in on branch but still wanting the flow to continue without user input? It may be simpler to just end the dialog at that point if there is nothing else to happen for the user. Or, if there is then to call another dialog that performs the additional actions you are expecting to occur.

**Comment by stevkan** [comment]
Closing due to inactivity. If the issue persists, please feel free to reopen.
